include/private/entry.h

```diff
@@ -27,24 +27,137 @@
 #  include <stumpless/severity.h>
 #  include "private/strbuilder.h"
 
+/**
+ * Frees entry cache
+ *
+ * **Thread Safety: MT-Unsafe**
+ * This function is not thread safe as it destroys resources(entry cache)
+ * that other threads can use.
+ *
+ * **Async Signal Safety: AS-Unsafe**
+ * This function is not safe to call from signal handlers due to destroying
+ * resources(entry cache) that might be used within a function.
+ *
+ * **Async Cancel Safety: AC-Unsafe**
+ * This function is not safe to call from threads that may be asynchronously
+ * cancelled, due to leaving memory in an undefined state.
+ *
+ * @since release v1.6.0
+ *
+ */
 void
 entry_free_all( void );
 
+/**
+ * Gets the priority value from facility and severity parameters.
+ * 
+ * Priority value is calculated by left shifting the facility value by 3 
+ * and adding it to the severity value which is according to the 
+ * RFC 5424 Section 6.2.1. The shift operation is done prior to get_prival()
+ * function with macros in "facility.h"
```

Comment:
This paragraph may be a bit confusing, as the facility `_VALUE` definitions are already shifted, and do not need to be shifted again. Since this is already taken care of elsewhere, it's probably best to simplify this paragraph to say something along the lines of "priority is calculated using the facility and severity values in accordance with RFC 5424 Section 6.2.1"

Reply:
I was a bit confused about this function as i didn't understand why it was calculating priority value that way so i took notes about the function inside the documentation and since i was confused at first i left it there in case someone wonders the same thing that i thought but it can be a bit confusing so i will shorten it as you said.